### PR TITLE
Re-ordered some human, Hai, Korath and Coalition ship hardpoints

### PR DESCRIPTION
**Pull request by warp-core**
Made all (I think) the human ships' hardpoints more consistent with the 'front to back, left to right' standard of hardpoint ordering.

### CODE REVIEW

**Comment by Bladewood** [comment]
coalition and shield beetle hardpoints too please? :P

**Comment by warp-core** [comment]
@Bladewood ok

**Comment by Bladewood** [comment]
yay~

**Comment by warp-core** [comment]
There a re some hard points that don't fit with this system on Korath ships as well so expect those to be changed tomorrow (hopefully).

File: data/ships.txt [hunk 1]
```diff
@@ -631,10 +631,10 @@ ship "Bulk Freighter"
 	gun -20 -175 "Sidewinder Missile Launcher"
 	gun 20 -175 "Sidewinder Missile Launcher"
 	turret -12 -155 "Laser Turret"
-	turret 30 175 "Laser Turret"
-	turret 0 5 "Laser Turret"
 	turret 12 -155 "Heavy Anti-Missile Turret"
 	turret -30 175 "Heavy Anti-Missile Turret"
+	turret 30 175 "Laser Turret"
+	turret 0 5 "Laser Turret"
```
Comment (endless-sky) [inline data/ships.txt]: Wrong order - this should be in the middle.

Comment (warp-core) [inline data/ships.txt]: True. I'll have to fix that tomorrow.

File: data/ships.txt [hunk 2]
```diff
@@ -963,16 +963,16 @@ ship "Cruiser"
...
-	gun -41 -52 "Particle Cannon"
-	gun 41 -52 "Particle Cannon"
 	gun -31 -63 "Particle Cannon"
 	gun 31 -63 "Particle Cannon"
+	gun -41 -52 "Particle Cannon"
+	gun 41 -52 "Particle Cannon"
 	gun -20 -44 "Sidewinder Missile Launcher"
 	gun 20 -44 "Sidewinder Missile Launcher"
 	turret 0 -40 "Heavy Laser Turret"
+	turret 0 0 "Heavy Anti-Missile Turret"
```
Comment (endless-sky) [inline data/ships.txt]: This should go back where it was. (0 > -19)

**Comment by endless-sky** [comment]
A lot of the ship variants (in "variants.txt") would also need updating to account for this. That's part of why I haven't bothered to make this change in the past.

(The story behind the odd hardpoints is that way back in the early days of the game, the saved game didn't store what hardpoints had what weapons, so the only way to make sure, say, that both ends of the Bulk Freighter had one laser and one anti-missile was to sort the hardpoints in the order that they currently have.)

**Comment by warp-core** [comment]
Yeah, it didn't occur to me about the variants until I was doing the Hai and Heliarch ships (hence why I changed the variants for those where necessary) so I'll do the variants tomorrow along with the Korath ships.

**Comment by warp-core** [comment]
@endless-sky Is there any reason I shouldn't reorder the variant ships to be in alphabetical order instead of the way it is now where they are grouped by category with each group alphabetized?

File: data/ships.txt [hunk 3]
```diff
@@ -688,18 +688,18 @@ ship "Carrier"
...
+	gun -25 -166 "Particle Cannon"
+	gun 25 -166 "Particle Cannon"
+	gun -25 -166 "Meteor Missile Launcher"
+	gun 25 -166 "Meteor Missile Launcher"
 	turret 0 -114 "Heavy Laser Turret"
-	turret 0 157 "Heavy Laser Turret"
-	turret 0 103 "Heavy Anti-Missile Turret"
+	turret 0 15 "Heavy Laser Turret"
+	turret 0 103 "Heavy Laser Turret"
+	turret 0 157 "Heavy Anti-Missile Turret"
```
Comment (Bladewood) [inline data/ships.txt]: Is this right?
In this version, the anti-missile is in the far back. But in the original, it was in between the second and third heavy laser turrets.

**Comment by Bladewood** [comment]
Make sure you don't forget the Alpha carrier variant~

**Comment by KiLEdEnNis** [comment]
@Bladewood ooh smart one

File: data/ships.txt (hunk 3 above)
Comment (warp-core) [inline data/ships.txt]: Good catch. Will fix.

**Comment by warp-core** [comment]
That should hopefully be everything.

**Comment by KiLEdEnNis** [comment]
@warp-core Marauders?

**Comment by warp-core** [comment]
@KiLEdEnNis those are their own unique definitions and not variants and they were added relatively recently so the hardpoints should be in the standard order but I'll check.

**Comment by KiLEdEnNis** [comment]
I checked too, good luck :P

**Comment by Bladewood** [comment]
I must now point out something that has been bothering me for quite a while.
https://github.com/endless-sky/endless-sky/blob/master/data/variants.txt#L544
The heavy laser splinter variant.
It's
[A] [A] [B]
A: heavy laser turret, B: heavy anti-missile
when it should be
[A] [B] [A]
The other one looks bad.

**Comment by KiLEdEnNis** [comment]
aah my autism ( ͡- _ʖ ͡-)

**Comment by Bladewood** [comment]
yay~

File: data/korath ships.txt [hunk 4]
```diff
@@ -1025,10 +1025,10 @@ ship "Model 64"
 	engine -19 159
 	engine 19 159
 	gun 0 -159 "Korath Minelayer"
-	turret -35 -69 "Korath Disruptor"
 	turret 20 -74 "Korath Slicer Turret"
-	turret 34 77 "Korath Disruptor"
 	turret -29 71 "Korath Slicer Turret"
+	turret -35 -69 "Korath Disruptor"
```
Comment (KiLEdEnNis) [inline data/korath ships.txt]: saw this line needs to move up a line, but I covered that in my PR that's upcoming

**Comment by Bladewood** [comment]
just remembered something, help

the Deep River has weird hardpoints

**Pull request merged**